test/addons-napi/test_promise/test.js

```diff
@@ -24,23 +25,25 @@ const assert = require('assert');
   promise.then(
     common.mustNotCall(),
     common.mustCall(function(result) {
-      assert.strictEqual(result, expected_result,
-                         `promise rejected as expected, received ${result}`);
+      assert.strictEqual(result, expected_result);
     }));
   test_promise.concludeCurrentPromise(expected_result, false);
 }
 
 // Chaining
-const promise = test_promise.createPromise();
-promise.then(
-  common.mustCall(function(result) {
-    assert.strictEqual(result, 'chained answer',
-                       'resolving with a promise chains properly');
-  }),
-  common.mustNotCall());
-test_promise.concludeCurrentPromise(Promise.resolve('chained answer'), true);
+{
+  const expected_result = 'chained answer';
+  const promise = test_promise.createPromise();
+  promise.then(
+    common.mustCall(function(result) {
+      assert.strictEqual(result, expected_result);
+    }),
+    common.mustNotCall());
+  test_promise.concludeCurrentPromise(Promise.resolve('chained answer'), true);
+
+  assert.strictEqual(test_promise.isPromise(promise), true);
```

Comment:
This one does not seem to belong here, can you move it back?

Reply:
@joyeecheung I suggested putting it in there because it uses `promise` because it was declared in that block. But looking more closely, perhaps we should create a separate promise object for this.

Reply:
Hmm yes, actually it should be `assert.strictEqual(test_promise.isPromise(test_promise.createPromise()), true);` then it can be safely moved down.

Reply:
I moved it out from the block, created new promise as @joyeecheung suggested